api/k8s/v1beta3/k8s_helpers.go

```diff
@@ -0,0 +1,49 @@
+package k8s
+
+import (
+	"time"
+
+	"github.com/scaleway/scaleway-sdk-go/internal/async"
+	"github.com/scaleway/scaleway-sdk-go/internal/errors"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+)
+
+// WaitForClusterRequest is used by WaitForCluster method.
+type WaitForClusterRequest struct {
+	ClusterID string
+	Region    scw.Region
+	Status    ClusterStatus
+	Timeout   time.Duration
+}
+
+// WaitForCluster wait for the cluster to be in a "terminal state" before returning.
+func (s *API) WaitForCluster(req *WaitForClusterRequest) (*Cluster, scw.SdkError) {
```

Comment:
method should return `error` if not this code would not work
```
_, err := WaitForCluster()
...
err = ioutils.ReadAll()
// Cannot assign error to scw.SdKError

```

Reply:
fixed